src/docs-website/src/theme/SearchBar/index.js

```
@@ -0,0 +1,391 @@
+    const activePlugin = useActivePlugin();
+    let versionUrl = baseUrl;
+    // For non-docs pages while using plugin-content-docs with custom ids,
+    // this will throw an error of:
+    //   > Docusaurus plugin global data not found for "docusaurus-plugin-content-docs" plugin with id "default".
+    // It seems that we can not get the correct id for non-docs pages.
+    try {
+        // The try-catch is a hack because useDocsPreferredVersion just throws an
+        // exception when versions are not used.
+        // The same hack is used in SearchPage.tsx
+        // eslint-disable-next-line react-hooks/rules-of-hooks
+        const { preferredVersion } = useDocsPreferredVersion(activePlugin?.pluginId ?? docsPluginIdForPreferredVersion);
```

Biome (2.4.4)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

(lint/correctness/useHookAtTopLevel)